### PR TITLE
Feature/retrolambda

### PR DESCRIPTION
**Pull request by tomrozb**
The optional config is in separate commit because it may not work for everyone. Please let me know if you cannot compile it so we can adjust or remove the optional config.

### CODE REVIEW

File: lib-android-common/build.gradle [hunk 1]
```diff
@@ -28,6 +34,14 @@ dependencies {
   provided dagger.dagger
 }
 
+retrolambda {
+  jdk System.getenv('JAVA8_HOME')
```
Comment (arturdm) [inline lib-android-common/build.gradle]: aren't those defaults?

File: lib-android-crashlytics/build.gradle [hunk 2]
```diff
@@ -37,6 +43,14 @@ dependencies {
   provided dagger.dagger
 }
 
+retrolambda {
+  jdk System.getenv('JAVA8_HOME')
```
Comment (arturdm) [inline lib-android-crashlytics/build.gradle]: defaults?

File: lib-android/build.gradle [hunk 3]
```diff
@@ -47,6 +53,14 @@ dependencies {
   androidTestCompile espressoCore
 }
 
+retrolambda {
+  jdk System.getenv('JAVA8_HOME')
```
Comment (arturdm) [inline lib-android/build.gradle]: defaults?

File: sample/build.gradle [hunk 4]
```diff
@@ -37,3 +43,11 @@ dependencies {
   apt dagger.compiler
   provided jsr205
 }
+
+retrolambda {
+  jdk System.getenv('JAVA8_HOME')
```
Comment (arturdm) [inline sample/build.gradle]: defaults?

File: lib-android-common/build.gradle (hunk 1 above)
Comment (tomrozb) [inline lib-android-common/build.gradle]: That's why it was a separate commit, now it's reverted.

**Pull request merged**